src/bin/mdbook-xgettext.rs

```diff
@@ -81,21 +90,242 @@ fn create_catalog(ctx: &RenderContext) -> anyhow::Result<Catalog> {
+                                    "{}:{}",
+                                    ctx.config.book.src.join(source).display(),
+                                    lineno
+                                );
+                                add_message(catalog, &msgid, &source2);
+                            });
+                    },
+                );
         }
     }
 
-    Ok(catalog)
+    Ok(catalogs)
+}
+
+// Contains information necessary to create polib messages out
+// of chapter content and save them into a template file.
+#[derive(Clone, Debug, Hash)]
+struct ChapterTemplateInfo {
+    // The chapter's content.
+    content: String,
+    // The file where the content is sourced.
+    source: String,
+    // The output destination for the messages
+    // created with the content and source.
+    template_destination: String,
+}
+
+// Get ChapterTemplateInfo for sub-chapters, depending on the specified depth.
+fn get_subcontent_for_chapter(
+    c: Chapter,
+    file_path: Option<String>,
+    depth: i64,
+) -> Vec<ChapterTemplateInfo> {
+    return extract_chapter_template_info(c, file_path, depth, 1);
+}
+
+// A recursive function to crawl a chapter's sub-items and get
+// the relevant info to produce a set of po template files.
+fn extract_chapter_template_info(
+    c: Chapter,
+    file_path: Option<String>,
+    depth: i64,
+    current_depth: i64,
+) -> Vec<ChapterTemplateInfo> {
+    if c.sub_items.is_empty() {
+        return Vec::new();
+    };
+
+    let mut vec = Vec::new();
+    c.sub_items.iter().for_each(|item| {
+        if let BookItem::Chapter(chapter) = item {
+            let file_path2 = match &chapter.path {
+                Some(rel_path) => {
+                    vec.push(ChapterTemplateInfo {
+                        content: chapter.content.clone(),       // content
+                        source: rel_path.display().to_string(), // source path (to save IN the template file)
+                        template_destination: file_path
+                            .clone()
+                            .unwrap_or(rel_path.display().to_string()), // where to save the template
+                    });
+                    let file_path2 = if current_depth + 1 > depth {
+                        file_path.clone()
+                    } else if current_depth + 1 == depth {
+                        Some(rel_path.display().to_string())
+                    } else {
+                        None
+                    };
+
+                    file_path2
+                }
+                None => file_path.clone(),
+            };
+            vec.append(&mut extract_chapter_template_info(
+                chapter.clone(),
+                file_path2,
+                depth,
+                current_depth + 1,
+            ));
+        }
+    });
+    vec
+}
+
+// Split a main catalog into sub-catalogs, using a provided split
+// depth to separate catalogs based on directory nesting.
+fn _split_catalog_on_depth(
```

Comment:
Identifiers starting with `_` are normally ignored in Rust — I don't know if this is still needed?